dds/DCPS/DataReaderImpl.cpp

```diff
@@ -143,8 +143,15 @@ DataReaderImpl::cleanup()
   // back onto the listener at the moment the related DDS entity has been
   // deleted
   set_listener(0, NO_STATUS_MASK);
+  topic_servant_ = 0;
```

Comment:
This is referenced in `~DataReaderImpl()`, which is causing the segfault in the ownership test.